plugins/base/src/main/kotlin/transformers/documentables/InheritorsExtractorTransformer.kt

```diff
@@ -0,0 +1,76 @@
+package org.jetbrains.dokka.base.transformers.documentables
+
+import org.jetbrains.dokka.links.DRI
+import org.jetbrains.dokka.model.*
+import org.jetbrains.dokka.model.Enum
+import org.jetbrains.dokka.model.properties.ExtraProperty
+import org.jetbrains.dokka.model.properties.MergeStrategy
+import org.jetbrains.dokka.pages.PlatformData
+import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.transformers.documentation.DocumentableTransformer
+
+class InheritorsExtractorTransformer : DocumentableTransformer, MapUtils {
+    override fun invoke(original: Module, context: DokkaContext): Module =
+        original.generateInheritanceMap().let { inheritanceMap -> original.appendInheritors(inheritanceMap) as Module }
+
+    private fun <T> T.appendInheritors(inheritanceMap: Map<PlatformData, Map<DRI, List<DRI>>>): Documentable where T : Documentable =
+        inheritanceMap.getForDRI(dri).toPlatformDependent().toInheritorsInfo().let { info ->
+            when (this) {
+                is Module -> copy(packages = packages.map { it.appendInheritors(inheritanceMap) as Package })
+                is Package -> copy(classlikes = classlikes.map { it.appendInheritors(inheritanceMap) as Classlike })
+                is Class -> copy(
+                    extra = extra + info,
+                    classlikes = classlikes.map { it.appendInheritors(inheritanceMap) as Classlike })
+                is Enum -> copy(
+                    extra = extra + info,
+                    classlikes = classlikes.map { it.appendInheritors(inheritanceMap) as Classlike })
+                is Interface -> copy(
+                    extra = extra + info,
+                    classlikes = classlikes.map { it.appendInheritors(inheritanceMap) as Classlike })
+                is Object -> copy(classlikes = classlikes.map { it.appendInheritors(inheritanceMap) as Classlike })
+                else -> this
```

Comment:
This `when` is lacking `Annotation` as subtyping annotation classes is possible in Java (unfortunately)